singularity-eos/eos/eos_spiner.hpp

```diff
@@ -26,19 +28,418 @@
+                                            Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  Real InternalEnergyFromDensityTemperature(const Real rho, const Real T,
+                                            Real *lambda = nullptr) const;
+
+  PORTABLE_INLINE_FUNCTION
+  Real PressureFromDensityTemperature(const Real rho, const Real T,
+                                      Real *lambda = nullptr) const;
+
+  PORTABLE_INLINE_FUNCTION
+  Real PressureFromDensityInternalEnergy(const Real rho, const Real sie,
+                                         Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  Real SpecificHeatFromDensityTemperature(const Real rho, const Real T,
+                                          Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  Real SpecificHeatFromDensityInternalEnergy(const Real rho, const Real sie,
+                                             Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  Real BulkModulusFromDensityTemperature(const Real rho, const Real T,
+                                         Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  Real BulkModulusFromDensityInternalEnergy(const Real rho, const Real sie,
+                                            Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  Real GruneisenParamFromDensityTemperature(const Real rho, const Real T,
+                                            Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  Real GruneisenParamFromDensityInternalEnergy(const Real rho, const Real sie,
+                                               Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  void DensityEnergyFromPressureTemperature(const Real press, const Real temp,
+                                            Real *lambda, Real &rho, Real &sie) const;
+  PORTABLE_INLINE_FUNCTION
+  void FillEos(Real &rho, Real &temp, Real &energy, Real &press, Real &cv, Real &bmod,
+               const unsigned long output, Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  void ValuesAtReferenceState(Real &rho, Real &temp, Real &sie, Real &press, Real &cv,
+                              Real &bmod, Real &dpde, Real &dvdt,
+                              Real *lambda = nullptr) const;
+
+  static constexpr unsigned long PreferredInput() { return _preferred_input; }
+  int matid() const { return matid_; }
+  PORTABLE_INLINE_FUNCTION Real lRhoOffset() const { return lRhoOffset_; }
+  PORTABLE_INLINE_FUNCTION Real lTOffset() const { return lTOffset_; }
+  PORTABLE_INLINE_FUNCTION Real lEOffset() const { return lEOffset_; }
+  PORTABLE_INLINE_FUNCTION Real rhoMin() const { return fromLog_(lRhoMin_, lRhoOffset_); }
+  PORTABLE_INLINE_FUNCTION Real rhoMax() const { return fromLog_(lRhoMax_, lRhoOffset_); }
+  PORTABLE_INLINE_FUNCTION Real TMin() const {
+    return fromLog_(T_.range(0).min(), lTOffset_);
```

Comment:
could these be initialized and stored at construction? I'm not sure how often these calls are made. I know this is unrelated to this PR, so maybe make an issue?

Reply:
yes they can and this is an easy fix. I can do it here.

Reply:
actually I'm gonna put this one off and make an issue.